src/vivarium/framework/time.py

```diff
@@ -14,14 +14,15 @@
 from __future__ import annotations
 
 import math
-from collections.abc import Callable
+from collections.abc import Callable, Sequence
 from functools import partial
 from typing import TYPE_CHECKING, Any
 
 import numpy as np
 import pandas as pd
 
 from vivarium.framework.lifecycle import lifecycle_states
+from vivarium.framework.resource import Resource
```

Comment:
Nit: Should you put this in the TYPE CHECKING block?

Reply:
We're not very strict about that, although I'd like to be.